src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java

```diff
@@ -1065,69 +1099,134 @@ private void biteBlock(@Nonnull BlockPos pos, @Nonnull IBlockState blockState, @
             NBTTagCompound updatePacket = bloodStats.writeUpdate(new NBTTagCompound());
             sync(updatePacket, true);
 
-            biteCooldown = Balance.vp.BITE_COOLDOWN;
 
         }
 
     }
 
+    /**
+     * Bite the entity with the given id.
+     * Checks reach distance
+     *
+     * @param entityId The id of the entity to start biting
+     */
+    public void biteEntity(int entityId) {
+        Entity e = player.getEntityWorld().getEntityByID(entityId);
+        if (player.isSpectator()) {
+            VampirismMod.log.w(TAG, "Player can't bite in spectator mode");
+            return;
+        }
+        if (e != null && e instanceof EntityLivingBase) {
+            if (e.getDistance(player) <= player.getEntityAttribute(EntityPlayer.REACH_DISTANCE).getAttributeValue() + 1) {
+                victim_bite_type = determineBiteType((EntityLivingBase) e);
+                if (victim_bite_type == BITE_TYPE.ATTACK || victim_bite_type == BITE_TYPE.ATTACK_HUNTER) {
+                    biteAttack((EntityLivingBase) e, victim_bite_type == BITE_TYPE.ATTACK_HUNTER);
+                } else if (victim_bite_type == BITE_TYPE.NONE) {
+                    return;
+                } else {
+                    if (victim == -1) biteTickCounter = 0;
+
+                    victim = e.getEntityId();
+                    PotionEffect effect = new PotionEffect(PotionBloodLoss.POTION, 20, 7, false, false);
+                    ((EntityLivingBase) e).addPotionEffect(effect);
+
+                    PotionEffect feedingEffect = new PotionEffect(PotionFeeding.POTION, 25, 1, false, false);
+                    player.addPotionEffect(feedingEffect);
+
+                    NBTTagCompound nbt = new NBTTagCompound();
+                    nbt.setInteger(KEY_VICTIM_ID, victim);
+                    sync(nbt, true);
+
+                }
+            } else {
+                VampirismMod.log.w(TAG, "Entity sent by client is not in reach " + entityId);
+            }
+        }
+    }
+
+
+    /**
+     * This is called every 20 ticks in onUpdate() to run the continuous feeding effect
+     */
+    public void biteVictim() {
+        EntityLivingBase e = ((EntityLivingBase) player.world.getEntityByID(victim));
```

Comment:
It is very unlikely, though still possible that getEntityByID returns an `Entity` instead of the expected `EntityLivingBase`.
In modded MC it is usually better to be safe than sorry :D